plugins/modules/azure_rm_registrationassignment.py

```diff
@@ -0,0 +1,381 @@
+        default: present
+        type: str
+        choices:
+            - absent
+            - present
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+author:
+    - GuopengLin (@t-glin)
+    - Fred-Sun (@Fred-Sun)
+    - Haiyuan Zhang (@haiyuazhang)
+
+'''
+
+EXAMPLES = '''
+    - name: Delete Registration Assignment
+      azure_rm_registrationassignment:
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        state: absent
+
+
+    - name: Put Registration Assignment
+      azure_rm_registrationassignment:
+        scope: subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        registration_assignment_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        properties:
+          registration_definition_id: /subscriptions/xxx-xxx/providers/Microsoft.ManagedServices/registrationDefinitions/xxx-xxx
+
+'''
+
+RETURN = '''
+properties:
+  description:
+      - Properties of a registration assignment.
+  returned: always
+  type: complex
+  contains:
+      registration_definition_id:
+          description:
+              - Fully qualified path of the registration definition.
+          returned: always
+          type: str
+          sample: null
+      registration_definition:
+          description:
+              - Registration definition inside registration assignment.
+          returned: always
+          type: complex
+          contains:
+              properties:
+                  description:
+                      - Properties of registration definition inside registration assignment.
+                  returned: always
+                  type: dict
+                  sample: null
+              plan:
+                  description:
+                      - Plan details for the managed services.
+                  returned: always
+                  type: complex
+                  contains:
+                      name:
+                          description:
+                              - The plan name.
+                          returned: always
+                          type: str
+                          sample: null
+                      publisher:
+                          description:
+                              - The publisher ID.
+                          returned: always
+                          type: str
+                          sample: null
+                      product:
+                          description:
+                              - The product code.
+                          returned: always
+                          type: str
+                          sample: null
+                      version:
+                          description:
+                              - The plan's version.
+                          returned: always
+                          type: str
+                          sample: null
+id:
```

Comment:
format is not correct, need intent?

Reply:
Id, properties, type and name are the values in the returned dictionary state. It increases the indentation. Thank you very much!